src/WebJobs.Extensions.CosmosDB/Bindings/CosmosDBItemValueBinder.cs

```diff
@@ -41,48 +40,35 @@ public async Task SetValueAsync(object value, CancellationToken cancellationToke
 
         public async Task<object> GetValueAsync()
         {
-            Uri documentUri = UriFactory.CreateDocumentUri(_context.ResolvedAttribute.DatabaseName, _context.ResolvedAttribute.CollectionName, _context.ResolvedAttribute.Id);
-            RequestOptions options = null;
+            T document = default(T);
 
-            if (!string.IsNullOrEmpty(_context.ResolvedAttribute.PartitionKey))
-            {
-                options = new RequestOptions
-                {
-                    PartitionKey = new PartitionKey(_context.ResolvedAttribute.PartitionKey)
-                };
-            }
-
-            Document document = null;
-
-            try
-            {
-                document = await _context.Service.ReadDocumentAsync(documentUri, options);
-            }
-            catch (DocumentClientException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
-            {
-                // ignore not found; we'll return null below
-            }
-
-            if (document == null)
-            {
-                return document;
-            }
-
-            T item = null;
+            PartitionKey partitionKey = _context.ResolvedAttribute.PartitionKey == null ? PartitionKey.None : new PartitionKey(_context.ResolvedAttribute.PartitionKey);
 
             // Strings need to be handled differently.
-            if (typeof(T) == typeof(string))
+            if (typeof(T) != typeof(string))
             {
-                _originalItem = JObject.FromObject(document);
-                item = _originalItem.ToString(Formatting.None) as T;
+                try
+                {
+                    document = await _context.Service.GetContainer(_context.ResolvedAttribute.DatabaseName, _context.ResolvedAttribute.CollectionName)
+                        .ReadItemAsync<T>(_context.ResolvedAttribute.Id, partitionKey);
+
+                    _originalItem = JObject.FromObject(document);
```

Comment:
I'm wondering if we want to continue supporting this "update-in-place" behavior. It's never worked for out-of-proc languages and has resulted in a decent amount of confusion as it seems all of the bindings handle it differently.

@fabiocav -- thoughts?